File: backend/app/services/processo_service.py
import uuid

from app.domain.models import Processo
from app.domain.schemas import ProcessoCreate
from app.repositories.processo_repo import ProcessoRepository


class ProcessoService:
    def __init__(self, repo: ProcessoRepository):
        self._repo = repo

    async def criar(self, dados: ProcessoCreate) -> Processo:
        processo = Processo(cliente_id=dados.cliente_id, nome=dados.nome)
        return await self._repo.criar(processo)

    async def obter(self, processo_id: uuid.UUID) -> Processo | None:
        return await self._repo.obter(processo_id)

    async def listar_por_cliente(self, cliente_id: uuid.UUID) -> list[Processo]:
        return await self._repo.listar_por_cliente(cliente_id)

    async def listar_todos(self) -> list[Processo]:
        return await self._repo.listar_todos()

File: frontend/src/app/core/api/services/processo.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

import { environment } from '../../../../environments/environment';
import { ProcessoCreate, ProcessoRead } from '../models/processo.model';

@Injectable({ providedIn: 'root' })
export class ProcessoApiService {
  private readonly baseUrl = `${environment.apiUrl}/processos`;

  constructor(private readonly http: HttpClient) {}

  listarPorCliente(clienteId: string): Observable<ProcessoRead[]> {
    const params = new HttpParams().set('cliente_id', clienteId);
    return this.http.get<ProcessoRead[]>(this.baseUrl, { params });
  }

  listarTodos(): Observable<ProcessoRead[]> {
    return this.http.get<ProcessoRead[]>(this.baseUrl);
  }

  obter(processoId: string): Observable<ProcessoRead> {
    return this.http.get<ProcessoRead>(`${this.baseUrl}/${processoId}`);
  }

  criar(dados: ProcessoCreate): Observable<ProcessoRead> {
    return this.http.post<ProcessoRead>(this.baseUrl, dados);
  }
}
